Add showTrend to Question

MartinodF opened the pull request:
This PR exposes the `showTrend` field on the Question object, allowing terraform users to read and update its value.
I've added basic tests for reading it (no update needed to cassettes, as they already contained it) and for validating it as a proper boolean.

While re-generating the GraphQL client and related docs to add `showTrend`, a couple of additional fields and values were also updated:
- `questionName` was removed from Rule
- `trigger_on_new_only` was added to Rule
- `FIFTEEN_MINUTES` was added as an acceptable value for SchedulerPollingInterval

Those changes are in a separate commit in case you want to exclude them.

chrichts commented on docs/resources/question.md:
```diff
@@ -39,6 +39,7 @@ resource "jupiterone_question" "unencrypted_critical_data_stores" {
 - `compliance` (Block List) (see [below for nested schema](#nestedblock--compliance))
 - `polling_interval` (String) Frequency of automated question evaluation. Defaults to ONE_DAY.
 - `query` (Block List) (see [below for nested schema](#nestedblock--query))
+- `show_trend` (Boolean) Whether to enable daily trend data collection. Defaults to false.
```
Maybe just update "daily trend data" to trend data, because its dependant on the polling interval and won't always be daily.

chrichts commented on jupiterone/internal/client/generated.go:
```diff
@@ -1184,14 +1184,15 @@ func (v *RuleStateInput) GetActions() map[string]interface{} { return v.Actions
-	SchedulerPollingIntervalTwelveHours   SchedulerPollingInterval = "TWELVE_HOURS"
-	SchedulerPollingIntervalOneDay        SchedulerPollingInterval = "ONE_DAY"
-	SchedulerPollingIntervalOneWeek       SchedulerPollingInterval = "ONE_WEEK"
+	SchedulerPollingIntervalDisabled       SchedulerPollingInterval = "DISABLED"
```
Let exclude this, its rule related and I know subject to change.

chrichts reviewed (changes requested):
@MartinodF - first off thank you for making these changes and sorry for taking so long to get back to you on this. If you have time, can you make the changes required, otherwise I can close this PR and resubmit a new PR with the required changes.

I've left 2 comments. To your point, lets exclude the PR making changes to the rule resource. This is subject to change again very soon. Besides the comments in the code, the following two functions also need to be updated (both in `resource_question.go`): `BuildQuestion` and `BuildCreateQuestionInput`. We need to add the `ShowTrend` variable to the question input. So the changes should look like:

```
func (qm *QuestionModel) BuildQuestion() client.QuestionUpdate {
	q := client.QuestionUpdate{
		Title:           qm.Title.ValueString(),
		Description:     qm.Description.ValueString(),
		Tags:            qm.Tags,
		ShowTrend:       qm.ShowTrend.ValueBool(),
		PollingInterval: client.SchedulerPollingInterval(qm.PollingInterval.ValueString()),
	}
``` 

and

```
func (qm *QuestionModel) BuildCreateQuestionInput() client.CreateQuestionInput {
	q := client.CreateQuestionInput{
		Title:           qm.Title.ValueString(),
		Description:     qm.Description.ValueString(),
		PollingInterval: client.SchedulerPollingInterval(qm.PollingInterval.ValueString()),
		ShowTrend:       qm.ShowTrend.ValueBool(),
		Tags:            qm.Tags,
	}
```

Let me know if you have any questions.

MartinodF commented:
Hey Chad, thank you for the feedback.

I've applied the requested changes:
- Removed changes related to the Rule resource
- Replaced "daily trend data" with "trend data"
- Provided `ShowTrend` value in both `CreateQuestionInput` and `QuestionUpdate`

Let me know if there's anything else that needs to be addressed!

chrichts commented:
Looks good thanks @MartinodF, one last thing and then I can approve, can you run a `go fmt` there is an extra white space in `generated.go`.

MartinodF commented:
You are right. I did run `make fmt`, then didn't add the whitespace change to the commit because it seemed unrelated and I couldn't figure out it was the formatter removing an extra space. Done 🙂
